railties/Rakefile

```diff
@@ -95,14 +95,18 @@ namespace :test do
       ])
       puts fake_command
 
-      # We could run these in parallel, but pretty much all of the
-      # railties tests already run in parallel, so ¯\_(⊙︿⊙)_/¯
-      Process.waitpid fork {
-        ARGV.clear.concat test_options
-        Rake.application = nil
-
-        load file
-      }
+      if defined?(JRUBY_VERSION)
```

Comment:
This check is performed for every `test_file` but we really only need to do it once.  We could define a method but IIRC doing that defines the method in the global scope. 

We could extract a class and define a method within it, but that felt a bit heavy handed for this script, didn't feel like this check was a particular perf bottleneck.

Reply:
Unfortunately, this ignores TruffleRuby (another alternative Ruby implementation). Much better and universal is `Process.respond_to?(:fork)`

Reply:
It might be better to add a fork feature check, but I don't think that truffle-ruby is there yet, so it LGTM as it is. Thanks for the PR!

Reply:
@ahorek who says it won't be in near future? I'm testing heavily TruffleRuby in production apps (142 reports in their issue tracker) and I wouldn't say that what you're claiming. This will be another problem preventing running full Rails tests in TruffleRuby.

Reply:
Checking for `fork` is indeed better here.

Reply:
> Much better and universal is Process.respond_to?(:fork)
❤️ 
I really like this, I was wondering how we could do this without having to maintain a list of `RIUBY_ENGINGE` or versions that don't support `fork`. 

I also mistakenly thought `JRUBY_VERSION` would actually be defined within TruffleRuby.

> I don't think that truffle-ruby is there yet

It's ok we'll be ready for when it is! Or do you mean something like `Process.respond_to?(:fork)` would actually break in `TruffleRuby`?

Reply:
No, it's the recommend in theirs [docs](https://github.com/oracle/truffleruby/blob/18c82f99b5b252ea6c18ed2604b2d3e280409ebc/doc/user/compatibility.md#fork) how to dealt with fork detection

Reply:
> Or do you mean something like Process.respond_to?(:fork) would actually break in TruffleRuby?

no, I can confirm `Process.respond_to?(:fork)` works fine.

Reply:
perfect, I've updated this PR, thanks!